fast_llm/models/gpt/model.py

```diff
@@ -72,34 +72,30 @@ def __init__(
             self._preprocessors.append(BackupAttentionPreprocessor(self._config.transformer, self._tensor_space))
 
     def get_output_layers(self) -> list[Layer]:
-        return [
-            layer
-            for i in range(self._config.prediction_heads)
-            for layer in [
-                TransformerLayer(
-                    self._config.transformer,
-                    self._tensor_space,
-                    # TODO MTP: which index?
-                    layer_index=self._config.transformer.num_layers,
-                    # The last layer only returns the transformer output.
-                    # The previous layers return a stack of shared_hidden and transformer_output.
-                    return_input=i < self._config.prediction_heads - 1,
-                ),
+        layers = []
```

Comment:
this rearrangement is slightly cleaner, good job!
I still don't understand why we care about the pathological case `num_layers = 0`, though.

Reply:
We care now because I'm using it in the test.

Reply:
Hmm, unfortunately this breaks for `prediction_heads>1` because we need `return_input=True` for the layer just before the first head.
Could we revert this change? @jlamypoirier

Reply:
Pushed a fix in. #235